lua/starfall/libs_sv/acffunctions.lua

```diff
@@ -1950,6 +1963,58 @@ function ents_methods:acfRoundType()
 	return BulletData and BulletData.Id or ""
 end
 
+--- Returns the BulletData table of the ammo in an ACF ammo crate
+-- @server
+function ents_methods:acfBulletData()
+	CheckType(self, ents_metatable)
+
+	local This = unwrap(self)
+
+	if not IsACFEntity(This) then SF.Throw("Entity is not valid", 2) end
+	if RestrictInfo(This) then return "" end
+
+	local BulletData = This.BulletData
+
+	return BulletData and shallowCopy(BulletData) or {}
```

Comment:
Does it not expose functions or anything?

Reply:
Not from my experience. Here is a sample output:

```
Id	=	AC
Crate	=	249
RoundVolume	=	49.32300466136
Efficiency	=	1
ProjMass	=	0.21641803472049
Diameter	=	2
ProjArea	=	3.1415926535898
ShovePower	=	0.2
Caliber	=	2
Type	=	AP
Tracer	=	0.3
PropLength	=	6.98
PropMass	=	0.020831900885954
Ricochet	=	60
DragCoef	=	0.0014516316339566
LimitVel	=	800
CartMass	=	0.23724993560645
Priority	=	Projectile
ProjLength	=	8.72
PropArea	=	3.1415926535898
MuzzleVel	=	454.92201677042
```